python/dgl/heterograph.py

```diff
@@ -2109,89 +2939,182 @@ def edge_ids(self, u, v, force_multi=None, return_uv=False, etype=None):
+            the edge with ID ``eid[i]``.
 
         Examples
         --------
         The following example uses PyTorch backend.
 
-        >>> g = dgl.bipartite(([0, 1, 1], [0, 0, 2]), 'user', 'plays', 'game')
-        >>> g.find_edges([0, 2], ('user', 'plays', 'game'))
-        (tensor([0, 1]), tensor([0, 2]))
-        >>> g.find_edges([0, 2])
-        (tensor([0, 1]), tensor([0, 2]))
+        >>> import dgl
+        >>> import torch
+
+        Create a homogeneous graph.
+
+        >>> g = dgl.graph((torch.tensor([0, 0, 1, 1]), torch.tensor([1, 0, 2, 3])))
+
+        Find edges of IDs 0 and 2.
+
+        >>> g.find_edges(torch.tensor([0, 2]))
+        (tensor([0, 1]), tensor([1, 2]))
+
+        For a graph of multiple edge types, it is required to specify the edge type in query.
+
+        >>> hg = dgl.heterograph({
+        >>>     ('user', 'follows', 'user'): (torch.tensor([0, 1]), torch.tensor([1, 2])),
+        >>>     ('user', 'plays', 'game'): (torch.tensor([3, 4]), torch.tensor([5, 6]))
+        >>> })
+        >>> hg.find_edges(torch.tensor([1, 0]), 'plays')
+        (tensor([4, 3]), tensor([6, 5]))
         """
+        if not (isinstance(eid, (numbers.Integral, Iterable)) or F.is_tensor(eid)):
+            raise DGLError('Expect eid to have type int, tensor or sequence, '
+                           'got {}'.format(type(eid)))
+
+        num_edges = self.num_edges(etype)
+
+        if isinstance(eid, numbers.Integral) and (eid < 0 or eid >= num_edges):
+            raise DGLError('Expect the edge ID to be from 0, ...'
+                           ', {:d}, got {:d}'.format(num_edges - 1, eid))
+
+        if not F.is_tensor(eid) and isinstance(eid, Iterable):
+            utils.detect_nan_in_iterable(eid, 'the edge IDs')
+            utils.detect_inf_in_iterable(eid, 'the edge IDs')
+
+        if F.is_tensor(eid) or isinstance(eid, Iterable):
+            utils.assert_nonnegative_iterable(eid, 'the edge IDs')
+            utils.assert_iterable_bounded_by_value(
+                eid, 'the edge IDs', num_edges,
+                'the number of {} edges'.format(etype))
+
         eid = utils.prepare_tensor(self, eid, 'eid')
         if len(eid) == 0:
             empty = F.copy_to(F.tensor([], self.idtype), self.device)
             return empty, empty
-        # sanity check
-        max_eid = F.as_scalar(F.max(eid, dim=0))
-        if max_eid >= self.number_of_edges(etype):
-            raise DGLError('Expect edge IDs to be smaller than number of edges ({}). '
-                           ' But got {}.'.format(self.number_of_edges(etype), max_eid))
         src, dst, _ = self._graph.find_edges(self.get_etype_id(etype), eid)
         return src, dst
 
     def in_edges(self, v, form='uv', etype=None):
-        """Return the inbound edges of the node(s) with the specified type.
+        """Return the incoming edges of some particular node(s) with the specified edge type.
```

Comment:
OK so we decided to use the term `incoming edges` and `outgoing edges`?  We need to revise our user guides and other docstrings accordingly.

Reply:
If you used "inbound" and "outbound", I can revert these.